src/MongoDB.Driver/Linq/Linq3Implementation/LinqProviderV3.cs

```diff
@@ -72,9 +72,22 @@ internal sealed class LinqProviderV3 : LinqProvider
             var context = new TranslationContext();
             var symbol = context.CreateSymbol(parameter, documentSerializer, isCurrent: true);
             context = context.WithSymbol(symbol);
-            var field = ExpressionToFilterFieldTranslator.Translate(context, expression.Body);
+            var body = RemovePossibleConvertToObject(expression.Body);
```

Comment:
This was the only change needed to get this scenario to work in LINQ3.

Reply:
`TranslateExpressionToField<TDocument>` removes the convert to object, but the overload `TranslateExpressionToField<TDocument, TField>` right below it does not. Is this the right level to be handling the removal of unnecessary converts or should these "tidy-ups" of the expression tree be done at the level of `ExpressionToFilterFieldTranslator.Translate` instead?